File: problems/problemset/mini-redis/Python/template/mini_redis.py
import socket

class MiniRedis:
    def __init__(self):
        pass

    def start(self, port: int) -> None:
        """Start a TCP server on the given port.
        Accept one connection at a time.
        For each connection, read RESP commands and write RESP responses.
        This method should block indefinitely."""
        pass

File: problems/problemset/mini-redis/Rust/template/mini_redis.rs
use std::net::TcpListener;

pub struct MiniRedis;

impl MiniRedis {
    pub fn new() -> Self {
        MiniRedis
    }

    pub fn start(&mut self, port: u16) {
        // Bind to the given port and accept connections one at a time.
        // For each connection, read RESP commands and write RESP responses.
        // This method should block indefinitely.
        let _listener = TcpListener::bind(("0.0.0.0", port)).unwrap();
        todo!()
    }
}
